### PR TITLE
added MAX31855 and A3967

### PR DESCRIPTION
**Pull request by mattcoates**
Added MAX31855 Thermocouple to Digital IC and the A3967 Stepper Motor Driver IC

**Datasheets:**
https://datasheets.maximintegrated.com/en/ds/MAX31855.pdf
https://cdn.sparkfun.com/datasheets/Robotics/A3967-Datasheet.pdf

### CODE REVIEW

File: lib/ic/power/a3967.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - [~ENABLE, 15, in]
+  - - [RC1, 23, in]
+    - [RC2, 2, in]
+    - [PFD, 24, in]
+- - - [VBB1, 20, pwrin]
+    - [VBB2, 5, pwrin]
+  - - [OUT1A, 16, pwrin]
+    - [OUT1B, 21, pwrin]
+  - - [OUT2A, 9, pwrin]
+    - [OUT2B, 4, pwrin]
```
Comment (adamgreig) [inline lib/ic/power/a3967.yaml:25]:
```suggestion
  - - [OUT1A, 16, out]
    - [OUT1B, 21, out]
  - - [OUT2A, 9, out]
    - [OUT2B, 4, out]
```

File: lib/ic/sensor/max31855.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+description: Cold-Junction Compensated Thermocouple-to-Digital Converter
+footprint: agg:SOIC-8
+ordercodes:
+  - [Farnell, '2515562']
+pins:
+- - - [VDD, 4, pwrin]
+    - [TC+, 3, in]
+    - [TC-, 2, in]
+    - [GND, 1, pwrin]
+- - - [SDO, 7, bidi]
```
Comment (adamgreig) [inline lib/ic/sensor/max31855.yaml:12]:
```suggestion
- - - [SDO, 7, out]
```

File: mod/ic/soic-24-w.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+chip_shape: [7.5, 15.4]
+description: SOIC-24-W
+model:
+  offset: [0, 0, 0]
+  path: ${KISYS3DMOD}/Package_SO.3dshapes/SOIC-24W_7.5x15.4mm_P1.27mm.wrl
+  rotate: [0, 0, 0]
+  scale: [1, 1, 1]
+name: SOIC-24-W
+pad_shape: [2.20, 0.60]
```
Comment (adamgreig) [inline mod/ic/soic-24-w.yaml:9]: Datasheet for A3967 has 0.65 wide pads, but 0.60 is probably fine too, was this deliberate?

Comment (mattcoates) [reply mod/ic/soic-24-w.yaml]: No this was not!

**Pull request merged**